src/main/java/net/sf/mzmine/modules/peaklistmethods/io/gnpsexport/GNPSSetupDialog.java

```diff
@@ -0,0 +1,325 @@
+package net.sf.mzmine.modules.peaklistmethods.io.gnpsexport;
+
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.GridBagConstraints;
+import java.awt.Window;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.Map;
+
+import javax.swing.Box;
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JComboBox;
+import javax.swing.JComponent;
+import javax.swing.JDialog;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+import javax.swing.text.JTextComponent;
+
+import net.sf.mzmine.main.MZmineCore;
+import net.sf.mzmine.parameters.Parameter;
+import net.sf.mzmine.parameters.ParameterSet;
+import net.sf.mzmine.parameters.UserParameter;
+import net.sf.mzmine.util.ExitCode;
+import net.sf.mzmine.util.GUIUtils;
+import net.sf.mzmine.util.components.GridBagPanel;
+
+public class GNPSSetupDialog extends JDialog
+implements ActionListener, DocumentListener {
+	
+	 private static final long serialVersionUID = 1L;
+
+    private ExitCode exitCode = ExitCode.UNKNOWN;
+
+    private String helpID;
+
+    // Parameters and their representation in the dialog
+    protected ParameterSet paramSet;
+    private final Map<String, JComponent> parametersAndComponents;
+
+    // If true, the dialog won't allow the OK button to proceed, unless all
+    // parameters pass the value check. This is undesirable in the BatchMode
+    // setup dialog, where some parameters need to be set in advance according
+    // to values that are not yet imported etc.
+    private final boolean valueCheckRequired;
+
+    // Buttons
+    private JButton btnOK, btnCancel;
+
+	
+	
+	protected GridBagPanel mainPanel;
+
+    /**
+     * Constructor
+     */
+    public GNPSSetupDialog(Window parent, boolean valueCheckRequired, ParameterSet parameters) {
```

Comment:
The issue with extending the class is that the parameters are protected and the addDialogComponents() needs to be overridden by the new class. Provided that most of the parameters are protected, I thought it was best to clone the class for now and then provide a better implementation by the next update.